### PR TITLE
[UST-193][Circuit] Improve Proof’s Readability

### PR DESCRIPTION
**Pull request by Lisooo790926**
## Summary
Add proof ts file to improve readability

## Linked Issue
[UST-193]

## Details
Based on new circuits to implement the proof class

## Impacted Areas
Current only impact on ReportIdentityProof

## Tests
Fix related tests to improve readability

## Suggestion 
Better put some comments in general purpose class to improve readability , thanks : )

### CODE REVIEW

**Comment by FoodChain1028** [comment]
Just want to remind this issue/ design idea from the [comment](https://github.com/social-tw/social-tw-website/pull/468#discussion_r1709793025) in PR #468

File: packages/circuits/src/ReportNegRepProof.ts
Comment (FoodChain1028) [inline packages/circuits/src/ReportNegRepProof.ts:1]: Hi @Lisooo790926 this is currently changed into `ReportNonNullifierProof` :) we should remove this and change into the latest version of proof. Thanks

Comment (Lisooo790926) [reply packages/circuits/src/ReportNegRepProof.ts:1]: Yes, I know, no worry, I will let you know once this PR is ready, thanks for reminding

**Comment by Lisooo790926** [comment]
@FoodChain1028 please check, and the relay test will fail randomly, it might not related to this PR.

**Review by FoodChain1028 — APPROVED** [review]
Looks perfect to me !!! Thank you!

**Pull request merged**